### ISSUE TITLE
@EnableRetry(proxyTargetClass = true) with CrudRepository

### ISSUE THREAD

**Issue by gmacster** [spring-projects/spring-retry#140]
### My issue

I'm getting the error message "Bean named 'myRepository' is expected to be of type 'com.mypackage.MyRepository' but was actually of type 'com.sun.proxy.$Proxy152'" when trying to setup a `@Retryable` annotation on one of my Spring Data `CrudRepository` interfaces, like so:

```java
@Retryable(TransientDataAccessException.class)
public interface MyRepository extends CrudRepository<MyEntity, Long> {

}
```

#### @EnableRetry(proxyTargetClass=true)

Based on some research, I set the `proxyTargetClass` to `true` on my `@EnableRetry` annotation to try and force it to use CGLIB proxies, but I'm still getting the same error. This is what my main application class looks like:

```java
@SpringBootApplication
@EnableRetry(proxyTargetClass = true)
public class MyApp {
  public static void main(String[] args) {
    SpringApplication.run(MyApp.class, args);
  }
}
```

#### @EnableAspectJAutoProxy(proxyTargetClass = true)

I noticed that the code for the `@EnableRetry` annotation imports a `@EnableAspectJAutoProxy` with `proxyTargetClass` hardcoded to `false` ([see here](https://github.com/spring-projects/spring-retry/blob/0753e4ee451d3b14f4bde7a4eb181140cb13bb8e/src/main/java/org/springframework/retry/annotation/EnableRetry.java#L40)), so I tried adding my own to my application but it had no effect:

```java
@SpringBootApplication
@EnableRetry(proxyTargetClass = true)
@EnableAspectJAutoProxy(proxyTargetClass = true)
public class MyApp {
  public static void main(String[] args) {
    SpringApplication.run(MyApp.class, args);
  }
}
```

I'm wondering if it's actually even possible to use the CGLIB proxies with Spring Retry, since that `@EnableAspectJAutoProxy` annotation is hardcoded _**not**_ to use them.

If anyone has any suggestions please let me know.

**Comment by gmacster** [comment]
I suppose I should mention I'm using spring-retry version 1.2.2.RELEASE...

**Comment by dsyer** [comment]
I wouldn't be surprised if Spring Data repositories were not proxyable. It wouldn't be a Spring Retry issue in that case right - it's a generic issue with Spring AOP and Spring Data? It is possible to use CGLib proxies with Spring Retry, but not with Spring Data because there is no base class for the target (so I'm not sure why you want to try to use them specifically). The configurability (or not) of `proxyTargetClass` is kind of a separate issue?

Bottom line - it's better to put the `@Retryable` in the caller. It might be awkward.
